libraries/terminal/src/StringBufferTerminalProvider.ts

```diff
@@ -19,6 +19,33 @@ export interface IStringBufferOutputOptions {
   normalizeSpecialCharacters: boolean;
 }
 
+/**
+ * @beta
+ */
+export interface IAllStringBufferOutput {
```

Comment:
This model is incomplete and allows drift in real-world output without breaking snapshots. We should really just buffer the raw chunks in sequence, merging consecutive chunks of the same severity, and run snapshots against that. It's rather important for our snapshot tests to capture the relative ordering of chunks of different severities, since that reflects what the user sees.